hosted/eks/helper/helper_cluster.go

```diff
@@ -84,12 +84,13 @@ func UpgradeClusterKubernetesVersion(cluster *management.Cluster, upgradeToVersi
 		}
 
 		// Check if the desired config has been applied in Rancher
-		Eventually(func() string {
-			ginkgo.GinkgoLogr.Info("Waiting for k8s upgrade to appear in EKSStatus.UpstreamSpec ...")
+		Eventually(func() bool {
+			ginkgo.GinkgoLogr.Info("Waiting for k8s upgrade to appear in EKSStatus.UpstreamSpec & EKSConfig ...")
 			cluster, err = client.Management.Cluster.ByID(cluster.ID)
 			Expect(err).To(BeNil())
-			return *cluster.EKSStatus.UpstreamSpec.KubernetesVersion
-		}, tools.SetTimeout(15*time.Minute), 30*time.Second).Should(Equal(upgradeToVersion))
+			return *cluster.EKSStatus.UpstreamSpec.KubernetesVersion == upgradeToVersion && *cluster.EKSConfig.KubernetesVersion == upgradeToVersion
```

Comment:
I don't think we need to check EKSConfig again.

Reply:
True, however `eks-operator/issues/668` occurs sometimes due to race condition.
This check ensures condition is met to start Nodegroup upgrade.

Reply:
Would you mind adding a comment about it if you haven't already? It's possible that I or someone else might forget later and remove it.